src/commands/apex/run/test.ts

```diff
@@ -0,0 +1,308 @@
+    // This was re-introduced due to https://github.com/forcedotcom/salesforcedx-vscode/issues/3154
+    // Address with W-9163533
+    if (flags.synchronous && testLevel === TestLevel.RunSpecifiedTests) {
+      result = await this.runTest(testService, flags, testLevel);
+    } else {
+      result = await this.runTestAsynchronous(testService, flags, testLevel);
+    }
+
+    if (this.cancellationTokenSource.token.isCancellationRequested) {
+      throw new SfError('Cancelled');
+    }
+
+    // check to make sure we have a TestResult
+    if ((result as TestResult).summary) {
+      result = result as TestResult;
+      const testReporter = new TestReporter(new Ux({ jsonEnabled: this.jsonEnabled() }), conn, this.config.bin);
+
+      return testReporter.report(result, {
+        wait: flags.wait,
+        'output-dir': flags['output-dir'],
+        'result-format': flags['result-format'],
+        'detailed-coverage': flags['detailed-coverage'],
+        synchronous: flags.synchronous,
+        json: flags.json,
+        codeCoverage: flags['code-coverage'],
+      });
+    } else {
+      // async test run
+      result = result as TestRunIdResult;
+      this.log(messages.getMessage('runTestReportCommand', [this.config.bin, result.testRunId, conn.getUsername()]));
+      return result;
+    }
+  }
+
+  // eslint-disable-next-line class-methods-use-this,complexity
+  public async validateFlags(
+    codeCoverage?: boolean,
+    resultFormatFlag?: string,
+    classNames?: string,
+    suiteNames?: string,
+    tests?: string,
+    synchronous?: boolean,
+    testLevel?: TestLevel
+  ): Promise<TestLevel> {
+    if (codeCoverage && !resultFormatFlag) {
+      return Promise.reject(new Error(messages.getMessage('missingReporterErr')));
+    }
+
+    if ((classNames && (suiteNames || tests)) || (suiteNames && tests)) {
+      return Promise.reject(new Error(messages.getMessage('classSuiteTestErr')));
+    }
+
+    if (synchronous && (suiteNames || (classNames && classNames.split(',').length > 1))) {
+      return Promise.reject(new Error(messages.getMessage('syncClassErr')));
+    }
+
+    if ((tests || classNames || suiteNames) && testLevel && testLevel !== 'RunSpecifiedTests') {
+      return Promise.reject(new Error(messages.getMessage('testLevelErr')));
+    }
+
+    let test: TestLevel;
+    if (testLevel) {
+      test = testLevel;
+    } else if (classNames || suiteNames || tests) {
+      test = TestLevel.RunSpecifiedTests;
+    } else {
+      test = TestLevel.RunLocalTests;
+    }
+
+    return test;
+  }
+
+  private async runTest(
+    testService: TestService,
+    flags: {
+      tests?: string;
+      'class-names'?: string;
+      'code-coverage'?: boolean;
+    },
+    testLevel: TestLevel
+  ): Promise<TestResult> {
+    const payload = await testService.buildSyncPayload(testLevel, flags.tests, flags['class-names']);
+    payload.skipCodeCoverage = !flags['code-coverage'];
```

Comment:
avoids mutating object props on a const
```suggestion
    const payload = {
       ...await testService.buildSyncPayload(testLevel, flags.tests, flags['class-names']), 
       skipCodeCoverage: !flags['code-coverage']
    }
```